### PR TITLE
openshift adm prune images

### PR DESCRIPTION
**Pull request by abikouo**
New module to prune images from Openshift cluster

### CODE REVIEW

**Comment by gravesm** [comment]
I'm still reviewing this (there's a lot here), but some quick feedback. I'd prefer we remove the docker-image-py dependency. We are pulling in a 3rd party library for what amounts to more or less a single regex. I think it's worth the extra effort to implement this ourselves and avoid the dependency.

**Comment by jillr** [comment]
Any time we pull in extra dependencies we need to work with the productization team to ensure they can be shipped in our execution environments.  I agree with @gravesm if it's avoidable we should do the extra work.

File: plugins/module_utils/openshift_adm_prune_images.py [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+
+    def prune_image_stream_tag(self, stream, tag_event_list):
+        manifests_to_delete, images_to_delete = [], []
+        filtered_items = []
+        tag_event_items = tag_event_list["items"] or []
+        prune_over_size_limit = self.params.get("prune_over_size_limit")
+        stream_namespace = stream["metadata"]["namespace"]
+        stream_name = stream["metadata"]["name"]
+        for idx in range(len(tag_event_items)):
+            item = tag_event_items[idx]
```
Comment (gravesm) [inline plugins/module_utils/openshift_adm_prune_images.py:316]: This appears in a few methods here, and while correct, it's a bit awkward to read. This is just a suggestion, but the following would probably be more readable:
```suggestion
        for idx, item in enumerate(tag_event_items):
```

Comment (abikouo) [reply plugins/module_utils/openshift_adm_prune_images.py:316]: will consider it 
Thanks

File: plugins/module_utils/openshift_import_image.py [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+            return True
+        self.find_resource(kind=kind, api_version=version, fail=True)
+        self.cluster_resources[key] = True
+
+    def create_image_import(self, ref):
+        kind = "ImageStream"
+        api_version = "image.openshift.io/v1"
+
+        # Validate that cluster has ImageStream Resource
+        self.test_cluster_resource(kind=kind, version=api_version)
```
[the remark this answers is not in the document]
Comment (abikouo) [reply plugins/module_utils/openshift_import_image.py:339]: resolved

**Comment by abikouo** [comment]
@gravesm Thanks for the feedback, the ``docker-image-py`` dependency will be removed.

**Comment by abikouo** [comment]
recheck

**Comment by goneri** [comment]
Add `coverage` in the `test-requirements.txt`.

**Comment by abikouo** [comment]
> Add `coverage` in the `test-requirements.txt`.

Thanks @goneri 
Will do this in a separated pull request

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Review by jillr — APPROVED** [review]
LGTM

**Review by gravesm — APPROVED** [review]
lgtm

**Pull request merged**